### PR TITLE
xbox: Change PCI IDs for USB and IDE controllers to match Xbox

### PR DESCRIPTION
**Pull request by GXTX**
I don't have my Xbox-Linux drive image (nor a working Xbox) any longer so I can't 100% confirm but going by my past tests should be enough. Although we are still missing 3 devices on the bus (memory controller, AC97 modem, pci-to-pci bridge).

Checking out my original patch (#130) on IDs, Matt had suggested getting a proper dump of the config space of all of the devices, someone should do that *for completeness*.

### CODE REVIEW

File: hw/usb/hcd-ohci-pci.c [hunk 1]
```diff
@@ -134,8 +134,14 @@ static void ohci_pci_class_init(ObjectClass *klass, void *data)
...
+    k->vendor_id = PCI_VENDOR_ID_NVIDIA;
+    k->device_id = PCI_DEVICE_ID_NVIDIA_NFORCE_USB;
+    k->revision  = 0xB1;
+#else
     k->vendor_id = PCI_VENDOR_ID_APPLE;
     k->device_id = PCI_DEVICE_ID_APPLE_IPID_USB;
+#endif
     k->class_id = PCI_CLASS_SERIAL_USB;
     set_bit(DEVICE_CATEGORY_USB, dc->categories);
     dc->desc = "Apple USB Controller";
```
Comment (binarymaster) [inline hw/usb/hcd-ohci-pci.c:150]:
```suggestion
#ifdef XBOX
    dc->desc = "nForce USB Controller";
#else
    dc->desc = "Apple USB Controller";
#endif
```

Comment (GXTX) [reply hw/usb/hcd-ohci-pci.c:150]: Is this presented to the OS? If so is that the correct description?

Comment (binarymaster) [reply hw/usb/hcd-ohci-pci.c:150]: Not sure, it might be some internal QEMU thing. Just suggested it for the consistence.

Comment (GXTX) [reply hw/usb/hcd-ohci-pci.c:150]: I dumped the USB reg area and didn't see any plain text like this. I assume this isn't passed onto the OS.

Comment (mborgerson) [reply hw/usb/hcd-ohci-pci.c:150]: It is used in device infrastructure (e.g. when printing out available devices). I agree with @binarymaster's suggested patch; Let's update the device description string while we are updating the id for consistency.

**Review by mborgerson — CHANGES_REQUESTED** [review]
See inline

File: hw/usb/hcd-ohci-pci.c (hunk 1 above)
Comment (GXTX) [reply hw/usb/hcd-ohci-pci.c:150]: Fixed

**Pull request merged**